test/compress/issue-203.js

```diff
@@ -26,9 +26,7 @@ compress_new_function_with_destruct: {
     }
     expect: {
         Function("a", "[b]", "return a");
-        Function("a", "{bb}", "return a");
-        Function("[[a]]", "[{bb}]", 'return a');
+        Function("a", "{bb:b}", "return a");
+        Function("[[a]]", "[{bb:b}]", 'return a');
```

Comment:
I'd rather have this `Function` optimization under its own flag `unsafe_Function` - with a capital F.
```
$ echo 'new Function("foo, {bar}", "return foo + bar;");' | bin/uglifyjs -c unsafe,side_effects=1
Function("foo, {bar}","return foo + bar;");

$ echo 'new Function("foo, {bar}", "return foo + bar;");' | bin/uglifyjs -c unsafe,side_effects=0
Function("a","{bar:b}","return a+b");
```
But the intention was probably for it to work with `unsafe=true` regardless of the `side_effects` setting. 

Ideally it should also respect the `mangle` option - which it does not have access to within `compress.js`.